src/draw/nxp/vglite/lv_draw_vglite_fill.c

```diff
@@ -89,18 +92,42 @@ void lv_draw_vglite_fill(lv_draw_task_t * t, const lv_draw_fill_dsc_t * dsc,
      * Most simple case: just a plain rectangle (no radius, no gradient)
      */
     if((dsc->radius == 0) && (dsc->grad.dir == (lv_grad_dir_t)LV_GRAD_DIR_NONE))
-        _vglite_fill(&clipped_coords, dsc);
+        _vglite_fill(vglite_task, &clipped_coords, dsc);
     else
-        _vglite_draw_rect(&relative_coords, &clip_area, dsc);
+        _vglite_draw_rect(vglite_task, &relative_coords, &clip_area, dsc);
 }
 
 /**********************
  *   STATIC FUNCTIONS
  **********************/
 
-static void _vglite_fill(const lv_area_t * dest_area, const lv_draw_fill_dsc_t * dsc)
+static void _vglite_set_rectangle(int32_t * path_data, uint32_t * path_data_size, const lv_area_t * dest_area)
+{
+    uint32_t pidx = 0;
+    path_data[pidx++] = VLC_OP_MOVE;
+    path_data[pidx++] = dest_area->x1;
+    path_data[pidx++] = dest_area->y1;
+    path_data[pidx++] = VLC_OP_LINE;
+    path_data[pidx++] = dest_area->x2 + 1;
+    path_data[pidx++] = dest_area->y1;
+    path_data[pidx++] = VLC_OP_LINE;
+    path_data[pidx++] = dest_area->x2 + 1;
+    path_data[pidx++] = dest_area->y2 + 1;
+    path_data[pidx++] = VLC_OP_LINE;
+    path_data[pidx++] = dest_area->x1;
+    path_data[pidx++] = dest_area->y2 + 1;
+    path_data[pidx++] = VLC_OP_LINE;
+    path_data[pidx++] = dest_area->x1;
+    path_data[pidx++] = dest_area->y1;
+    path_data[pidx] = VLC_OP_END;
+
+    *path_data_size = pidx * sizeof(int32_t);
```

Comment:
`path_data_size` is set to 4 less than the actual.

```suggestion
    path_data[pidx++] = dest_area->y1;
    path_data[pidx++] = VLC_OP_END;

    *path_data_size = pidx * sizeof(int32_t);
```